### ISSUE TITLE
nvim-ts-autotag does not work with null-ls formatting

### ISSUE THREAD

**Issue by kwkarlwang** [windwp/nvim-ts-autotag#44]
Hi windxp,

I have notice that nvim-ts-autotag will stop working if I use null-ls to format. I have raise the issue at [null-ls](https://github.com/jose-elias-alvarez/null-ls.nvim/issues/626).

## Minimal reproducible config

```lua
local on_windows = vim.loop.os_uname().version:match("Windows")

local function join_paths(...)
    local path_sep = on_windows and "\\" or "/"
    local result = table.concat({ ... }, path_sep)
    return result
end

vim.cmd([[set runtimepath=$VIMRUNTIME]])

local temp_dir
if on_windows then
    temp_dir = vim.loop.os_getenv("TEMP")
else
    temp_dir = "/tmp"
end

vim.cmd("set packpath=" .. join_paths(temp_dir, "nvim", "site"))

local package_root = join_paths(temp_dir, "nvim", "site", "pack")
local install_path = join_paths(package_root, "packer", "start", "packer.nvim")
local compile_path = join_paths(install_path, "plugin", "packer_compiled.lua")

local null_ls_config = function()
    local null_ls = require("null-ls")
    -- add only what you need to reproduce your issue
    null_ls.setup({
        sources = {
            null_ls.builtins.formatting.prettier
        },
        debug = true,
    })
end

local function load_plugins()
    -- only add other plugins if they are necessary to reproduce the issue
    require("packer").startup({
        {
            "wbthomason/packer.nvim",
            {
                "jose-elias-alvarez/null-ls.nvim",
                requires = { "nvim-lua/plenary.nvim" },
                config = null_ls_config,
            },{
                'nvim-treesitter/nvim-treesitter',
                run = ':TSUpdate'
            },{
                "windwp/nvim-ts-autotag",
                config = function ()
                    require('nvim-ts-autotag').setup()
                end
            }
        },
        config = {
            package_root = package_root,
            compile_path = compile_path,
        },
    })
end

if vim.fn.isdirectory(install_path) == 0 then
    vim.fn.system({ "git", "clone", "https://github.com/wbthomason/packer.nvim", install_path })
    load_plugins()
    require("packer").sync()
else
    load_plugins()
    require("packer").sync()
end
```

## Steps to reproduce

In short, nvim-ts-autotag does not close the tag after calling `lua vim.lsp.buf.formatting_sync()` **manully**.

1. After installing the plugins. Run `:TSInstall tsx`
2. Run the following command: `mkdir test && cd test && npm init -y` in terminal
3. Create a new file `nvim index.tsx` and paste the following content
```tsx
import React from 'react'
export const index = (props : {}) => {
    return;
}
```
4. Inside neovim, run `:lua vim.lsp.buf.formatting_sync()`.
5. In between `return` and `;`, type ` <div>`.

Here is a video to demonstrate the problem.

https://user-images.githubusercontent.com/38927155/151673509-7fcc302c-3b90-45ac-addb-71502e66c382.mov

Based on @jose-elias-alvarez comments in the issue, I have made a fork of nvim-ts-autotag that fixes the problem. I wonder if you are interested in a PR that fixes this problem?

**Comment by windwp** [comment]
you can make a PR.
Thank

**Comment by kwkarlwang** [comment]
Fixed in #45
